raft.c

```diff
@@ -1575,21 +1589,33 @@ static void handleCfgChange(RedisRaftCtx *rr, RaftReq *req)
     entry->id = rand();
     entry->type = type;
     memcpy(entry->data, &req->r.cfgchange, sizeof(req->r.cfgchange));
+    entryAttachRaftReq(rr, entry, req);
 
-    if ((e = raft_recv_entry(rr->raft, entry, &response)) != 0) {
+    e = raft_recv_entry(rr->raft, entry, &response);
+    if (e != 0) {
+        entryDetachRaftReq(rr, entry);
+        raft_entry_release(entry);
         replyRaftError(req->ctx, e);
-    } else {
-        if (req->type == RR_CFGCHANGE_REMOVENODE) {
-            RedisModule_ReplyWithSimpleString(req->ctx, "OK");
-        } else {
+        goto exit;
+    }
+
+    raft_entry_release(entry);
+    switch (req->type) {
+        case RR_CFGCHANGE_ADDNODE:
+            // we don't have to block on add node, its all through join which blocks itself
+            entryDetachRaftReq(rr, entry);
             RedisModule_ReplyWithArray(req->ctx, 2);
             RedisModule_ReplyWithLongLong(req->ctx, req->r.cfgchange.id);
             RedisModule_ReplyWithSimpleString(req->ctx, rr->snapshot_info.dbid);
-        }
+            break;
+        case RR_CFGCHANGE_REMOVENODE:
+            // block until removed, so don't reply here
+            return;
+        default:
+            replyRaftError(req->ctx, e);
```

Comment:
I think it can't fall into "default" case. Even it will somehow, 'e' is 0 always. Should we use assert(0) instead or just if check as if before?